src/Routing/CreateUIRouteProvider.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\entity\Routing\CreateUIRouteProvider.
+ */
+
+namespace Drupal\entity\Routing;
+
+use Drupal\Core\Entity\EntityTypeInterface;
+use Drupal\Core\Entity\Routing\EntityRouteProviderInterface;
+use Symfony\Component\Routing\Route;
+use Symfony\Component\Routing\RouteCollection;
+
+/**
+ * Additional common routes needed for a CRUD UI.
+ *
+ * - add bundle add overview page (like /node/add)
+ * - a generic bundle specific entity add form (like /node/add/page)
+ */
+class CreateUIRouteProvider implements EntityRouteProviderInterface {
```

Comment:
This route provider sounds like it's completing the work of DefaultHtmlRouteProvider, it's not just adding create routes, it's also adding the collection one. So maybe we give it a more generic name? ExtendedHtmlRouteProvider? ContentEntityRouteProvider?